### ISSUE TITLE
Dupe with Danknull installed

### ISSUE THREAD

**Issue by Freebi85** [codetaylor/artisan-worktables#227]
Hey my friend this is a weird one ó_Ò

After the placement of a workshop in survival mode an infinite amount of danknulls tier 2 can be obtained from the output slot. This happens with any kind of Workshop. Also with those without any crafttweaker scripts/recipes.

This happen on the Dungeons Dragons and space shuttles modpack while testing latest version of Artisan tables and Dank Null.

* Minecraft: 1.12.2
* Forge: 14.23.5.2847
* CraftTweaker: 4.1.20.562
* Artisan Worktables: 1.22.17
* Athenaeum: 1.17.4
* Dank Null: 1.7.94

**Comment by codetaylor** [comment]
Thanks for the report! This is fixed in `1.22.18`

**Comment by codetaylor** [comment]
Please don't use `1.22.18`. I rushed it and introduced a new bug.

**Comment by codetaylor** [comment]
Please use version `1.22.19` as it has zero known dupe bugs. XD

**Comment by codetaylor** [comment]
`1.22.19` broke vanilla recipe matching, please use `1.22.20` ...

It's a f'n circus up in here.

**Comment by Freebi85** [comment]
I'm so sorry to bother you again but sadly this dupe is still a thing :(

After crafting a few items it workes fine. But suddenly all workshops outputs dank nulls again for some reason :(

it's hard to give you some exact steps to reproduce because it doesent happen immediatly. After it happens in a specific world all Workshops have same problem.
Not sure if this can be done on your end at all.

Edit: Seems to hapen after crafting something with fluids until the fluid gets consumed completely and then moving items around. Still investigating

**Comment by codetaylor** [comment]
Wuuut. Ok, no bother. :)

What version are you on now?

**Comment by codetaylor** [comment]
I think I know what's causing it.

If you legit craft a dank null upgrade, then it gets cached as the last valid recipe. The first recipe checked is the cached recipe. In this case, it's caching the dank null recipe and checking an empty grid against that recipe.

It's a flaw in the dank null upgrade recipe matching and I've submitted a PR to fix it, linked above.

I can work around this however, as the table shouldn't be trying to match any recipes when its crafting grid is empty.

No worries, Freebi, I got this. :)
